services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java

```diff
@@ -1,52 +1,337 @@
+    assertEquals("ab", names.get(0));
+    assertEquals(null, names.get(1));
+    assertEquals(null, names.get(2));
+    assertEquals("cd", names.get(3));
+    assertEquals("ef", names.get(4));
+    assertEquals(null, names.get(5));
+    assertEquals(null, names.get(6));
+    assertEquals("gh", names.get(7));
+    assertEquals("ij", names.get(8));
+  }
+
+  @Test
+  public void waypointNamesList() {
+    List<String> names = new ArrayList<>();
+    names.add(null);
+    names.add(null);
+    names.add("abc");
+    names.add(null);
+    names.add("def");
+    names.add(null);
+    names.add(null);
+
+    RouteOptions routeOptions = routeOptions()
+        .toBuilder()
+        .waypointNamesList(names)
+        .build();
+
+    assertEquals(names, routeOptions.waypointNamesList());
+    assertEquals(";;abc;;def;;", routeOptions.waypointNames());
+  }
+
+  @Test
+  public void waypointTargetsString() {
+    String targetsStr = "1.2,3.4;;;5.65,7.123;;;";
+
+    RouteOptions routeOptions = routeOptions()
+        .toBuilder()
+        .waypointTargets(targetsStr)
+        .build();
+
+    assertEquals("1.2,3.4;;;5.65,7.123", routeOptions.waypointTargets());
+
+    List<Point> targets = routeOptions.waypointTargetsList();
+    assertEquals(4, targets.size());
+    assertEquals(Point.fromLngLat(1.2, 3.4), targets.get(0));
+    assertEquals(null, targets.get(1));
+    assertEquals(null, targets.get(2));
+    assertEquals(Point.fromLngLat(5.65, 7.123), targets.get(3));
+  }
+
+  @Test
+  public void waypointTargetsList() {
+    List<Point> targets = new ArrayList<>();
+    targets.add(null);
+    targets.add(null);
+    targets.add(Point.fromLngLat(5.55, 7.77));
+    targets.add(Point.fromLngLat(1.22, 3.44));
+    targets.add(null);
+    targets.add(Point.fromLngLat(1.55, 8.99));
+
+    RouteOptions routeOptions = routeOptions()
+        .toBuilder()
+        .waypointTargetsList(targets)
+        .build();
+
+    assertEquals(targets, routeOptions.waypointTargetsList());
+    assertEquals(";;5.55,7.77;1.22,3.44;;1.55,8.99", routeOptions.waypointTargets());
+  }
+
+  @Test
+  public void annotationsString() {
+    String annotationsStr = ANNOTATION_MAXSPEED + ";" + ANNOTATION_DURATION + ";" + ";";
+
+    RouteOptions routeOptions = routeOptions()
+        .toBuilder()
+        .annotations(annotationsStr)
+        .build();
+
+    assertEquals(ANNOTATION_MAXSPEED + ";" + ANNOTATION_DURATION, routeOptions.annotations());
+
+    List<String> annotations = routeOptions.annotationsList();
+    assertEquals(2, annotations.size());
+    assertEquals(ANNOTATION_MAXSPEED, annotations.get(0));
+    assertEquals(ANNOTATION_DURATION, annotations.get(1));
+  }
+
+  @Test
+  public void annotationsList() {
+    List<String> annotations = new ArrayList<>();
+    annotations.add(ANNOTATION_CONGESTION);
+    annotations.add(ANNOTATION_DISTANCE);
+    annotations.add(ANNOTATION_MAXSPEED);
+    annotations.add(ANNOTATION_SPEED);
+    annotations.add(null);
```

Comment:
Not sure if we want to accept `null` and then trim them 🤔 But it's good for now.

Reply:
passed Annotations List might have `null` values. I think we can't forbid it, so we need to have logic to remove them. Also we can update docs (in a separate PR) to make it valid params clearer.